### PR TITLE
Add warning when content to upload exceeds limit

### PR DESCRIPTION
**Pull request by iitzco-asapp**
(cc @ciguaran-asapp)

When uploading heavy content to the cluster (either an extension or a resource), `flambe` was lacking a warning so that users understand where this delay is coming from.

This PR fixes this.

- [x] Add unit testing for the `get_size_MB` method

### CODE REVIEW

File: flambe/runner/utils.py [hunk 1]
```diff
@@ -26,11 +26,14 @@ def get_folder_size_MB(path: str) -> float:
...
-        for f in filenames:
-            fp = os.path.join(dirpath, f)
-            if os.path.exists(fp) and not os.path.islink(fp):
-                accum += os.path.getsize(fp)
+    if os.path.isdir(path):
+        for dirpath, dirnames, filenames in os.walk(path):
+            for f in filenames:
+                fp = os.path.join(dirpath, f)
+                if os.path.exists(fp) and not os.path.islink(fp):
+                    accum += os.path.getsize(fp)
```
Comment (ciguaran-asapp) [inline flambe/runner/utils.py]: Do we really need to do a summation here? It shouldn't be necessary to traverse the path.

File: flambe/runner/utils.py [hunk 2]
```diff
@@ -50,6 +53,6 @@ def check_system_reqs() -> None:
 
     # Check if extensions folder is getting big
     extensions_folder = os.path.join(FLAMBE_GLOBAL_FOLDER, "extensions")
-    if get_folder_size_MB(extensions_folder) > WARN_LIMIT_MB:
+    if os.path.exists(extensions_folder) and get_size_MB(extensions_folder) > WARN_LIMIT_MB:
```
Comment (ciguaran-asapp) [inline flambe/runner/utils.py]: I don't follow why we are logging this in two different places.

File: flambe/runner/utils.py (hunk 1 above)
Comment (iitzco-asapp) [reply flambe/runner/utils.py]: There is no standard way of getting the size of a directory (something similar to `du`). `os.walk` seems the way to go here, even the `os.walk` docs has an example on how to do this (https://docs.python.org/3/library/os.html#os.walk)

File: flambe/runner/utils.py (hunk 2 above)
Comment (iitzco-asapp) [reply flambe/runner/utils.py]: Good point.

File: flambe/cluster/cluster.py [hunk 3]
```diff
@@ -571,6 +575,11 @@ def send_local_content(self,
             c = os.path.expanduser(c)
             base: str = ""
             if os.path.exists(c):
+
+                size = get_size_MB(c)
+                if size > UPLOAD_WARN_LIMIT_MB:
+                    logger.info(cl.YE(f"Uploading may take time for ({int(size)} MB) '{c}'"))
```
Comment (nmatthews-asapp) [inline flambe/cluster/cluster.py]:
```suggestion
                    logger.info(cl.YE(f"Uploading '{c}' ({int(size)} MB) which may take a while; if this is a surprise, double check you want to be transferring this file. Note we automatically sync resources, extensions (and any files in your flambe installation if installed with '-e')"))
```

File: flambe/cluster/cluster.py [hunk 4]
```diff
@@ -571,6 +575,15 @@ def send_local_content(self,
             c = os.path.expanduser(c)
             base: str = ""
             if os.path.exists(c):
+
+                size = get_size_MB(c)
+                if size > UPLOAD_WARN_LIMIT_MB:
+                    logger.info(cl.YE(
+                        f"Uploading '{c}' ({int(size)} MB) which may take a while. " +
+                        "Double check you want to be transferring this file " +
+                        "(note we automatically sync extensions and experiment resources " +
```
Comment (iitzco-asapp) [inline flambe/cluster/cluster.py]:
```suggestion
                        "(note we automatically sync extensions, experiment resources " +
```

File: flambe/cluster/cluster.py [hunk 5]
```diff
@@ -571,6 +575,15 @@ def send_local_content(self,
...
+                        "and potentially the flambe repo if installed in dev mode)"))
```
Comment (iitzco-asapp) [inline flambe/cluster/cluster.py]:
```suggestion
                        "and the flambe repo if installed in dev mode)"))
```

**Pull request merged**